src/application/sidebar/FileExplorer.tsx

```diff
@@ -8,41 +8,63 @@ import {
 } from '../../atoms/fileExplorerActions';
 import { useActiveEditorIdForPane } from '../../atoms/hooks/useActiveEditorIdForPane';
 import { parseEditorId } from '../../atoms/types/EditorData';
-import { FileExplorerFileEntry, FileExplorerFolderEntry } from '../../atoms/types/FileExplorerEntry';
+import { FileExplorerEntry, FileExplorerFileEntry, FileExplorerFolderEntry } from '../../atoms/types/FileExplorerEntry';
 import { FileNameInput } from '../../components/FileNameInput';
 import { emitEvent } from '../../events';
 import { useAsyncEffect } from '../../hooks/useAsyncEffect';
+import { isInUploadsDir } from '../../io/filesystem';
 import { cx } from '../../lib/cx';
 import { isNonNullish } from '../../lib/isNonNullish';
 import { FILE_EXPLORER_FILE_MENU_ID } from './fileExplorerContextMenu/FileExplorerFileContextMenu';
-import { useFileExplorerContextMenu } from './fileExplorerContextMenu/useFileExplorerContextMenu';
-import { ArrowRightIcon } from './icons';
+import { FILE_EXPLORER_PROJECT_MENU_ID } from './fileExplorerContextMenu/FileExplorerProjectContextMenu';
+import {
+  useFileExplorerContextMenu,
+  useFileExplorerOptionalContextMenu,
+} from './fileExplorerContextMenu/useFileExplorerContextMenu';
+import { ArrowDownIcon, HorizontalDotsIcon } from './icons';
+
+export function FileExplorer({ projectName }: { projectName: string }) {
+  const [collapsed, setCollapsed] = useState(false);
```

Comment:
This is stored in an atom for projects folders, so that when you close and re-open the explorer, everything kept its state. I think it would be great to also save that in an atom for the projects.
I think we should create an atom `openProjectState` with some useful values like `name`, `id` and `isCollapsed`. When we update the app so support multiple projects, we can then update the atom to be a list of `ProjectState`s

Reply:
I see your idea, but I think we should implement that later once we really support having multiple projects open at the same time. For now we are only making sure we have the design ready.